app/src/main/java/org/piwigo/io/repository/PreferencesRepository.java

```diff
@@ -51,4 +52,13 @@ private void set(String key, String value) {
         editor.putString(key, value);
         editor.apply();
     }
+
+    public String getStringPreference(String key, String value) {
+        return preferences.getString(key, value);
+    }
+
+    public Integer getIntegerPreference(String key, String value) {
```

Comment:
Dont know, as you wish. What are the benefits of int only ? :D

Reply:
int is more lightweight, as it is a primitive datatype in the java virtual machine instead of an object containing an int, so it can be passed around much more easy (technically on low level) than an Integer which is even sitting in the Heap...

Reply:
And furthermore you should not set a string for an int value and parse it for each extraction but use preferences.getInt() instead. In this specific case where you use a list for the different integer values you might need parsing a String, but that shall be done during inserting the value not every time we need it - which will be for each and every album view.

Reply:
![image](https://user-images.githubusercontent.com/56294284/68330959-683b3980-00d4-11ea-930b-f83934a16931.png) I was not able to change that in SettingsActivity Listener. I used https://stackoverflow.com/questions/3661413/how-to-cast-an-object-to-an-int

Reply:
The ListPreference seemingly sets and expects only Strings. What about changing it to SeekBarPreference? - I submitted a PR against your repo with a proposal...

Reply:
Ok will try